### PR TITLE
tempo: 1.1.0 -> 1.4.1, add NixOS module

### PR DESCRIPTION
**Pull request by K900**
###### Description of changes

Catching up on nearly a year of updates.
Changelogs: https://github.com/grafana/tempo/releases

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/tracing/tempo.nix [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  inherit (lib) escapeShellArgs mkEnableOption mkIf mkOption types;
+
+  cfg = config.services.tempo;
+
+  settingsFormat = pkgs.formats.yaml {};
+in {
+  options.services.tempo = {
+    enable = mkEnableOption "Grafana Tempo";
+
+    configuration = mkOption {
```
Comment (SuperSandro2000) [inline nixos/modules/services/tracing/tempo.nix:13]:
```suggestion
    settings = mkOption {
```
Most other modules name it settings.

File: nixos/modules/services/tracing/tempo.nix [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    # for tempo-cli and friends
+    environment.systemPackages = [ pkgs.tempo ];
+
+    assertions = [{
+      assertion = (
+        (cfg.configuration == {} -> cfg.configFile != null) &&
+        (cfg.configFile != null -> cfg.configuration == {})
+      );
+      message  = ''
+        Please specify either
```
Comment (SuperSandro2000) [inline nixos/modules/services/tracing/tempo.nix:40]:
```suggestion
        Please specify a tempo configuration with either
```

File: nixos/modules/services/tracing/tempo.nix [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+  cfg = config.services.tempo;
+
+  settingsFormat = pkgs.formats.yaml {};
+in {
+  options.services.tempo = {
+    enable = mkEnableOption "Grafana Tempo";
+
+    configuration = mkOption {
+      type = (pkgs.formats.json {}).type;
```
Comment (pennae) [inline nixos/modules/services/tracing/tempo.nix:14]:
```suggestion
      type = settingsFormat.type;
```

File: nixos/modules/services/tracing/tempo.nix [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+  config = mkIf cfg.enable {
+    # for tempo-cli and friends
+    environment.systemPackages = [ pkgs.tempo ];
+
+    assertions = [{
+      assertion = (
+        (cfg.configuration == {} -> cfg.configFile != null) &&
+        (cfg.configFile != null -> cfg.configuration == {})
+      );
```
Comment (pennae) [inline nixos/modules/services/tracing/tempo.nix:38]:
```suggestion
      assertion = (cfg.configuration == {}) != (cfg.configFile != null);
```
probably easier to read considering how many people we've seen confused by `->`

File: pkgs/servers/tracing/tempo/default.nix [hunk 5]
```diff
@@ -1,18 +1,28 @@
...
 
 buildGoModule rec {
-  version = "1.1.0";
+  version = "1.4.1";
   pname = "tempo";
 
   src = fetchFromGitHub {
     rev = "v${version}";
     owner = "grafana";
     repo = "tempo";
```
Comment (SuperSandro2000) [inline pkgs/servers/tracing/tempo/default.nix:10]:
```suggestion
    owner = "grafana";
    repo = "tempo";
    rev = "v${version}";
```

File: pkgs/servers/tracing/tempo/default.nix [hunk 6]
```diff
@@ -1,18 +1,28 @@
...
-    sha256 = "sha256-qKsgcc62HTwl7Usmp8zk4vKDo4XEJnwL+A3hoLhgBkk=";
+    sha256 = "sha256-kxR+xwhthsK3gThs0jPJfWlsRG35kCuWvKH3Wr7ENTs=";
+    fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/servers/tracing/tempo/default.nix:12]:
```suggestion
    fetchSubmodules = true;
    sha256 = "sha256-kxR+xwhthsK3gThs0jPJfWlsRG35kCuWvKH3Wr7ENTs=";
```

File: pkgs/servers/tracing/tempo/default.nix [hunk 7]
```diff
@@ -1,18 +1,28 @@
 { lib, buildGoModule, fetchFromGitHub }:
 
 buildGoModule rec {
-  version = "1.1.0";
+  version = "1.4.1";
   pname = "tempo";
```
Comment (SuperSandro2000) [inline pkgs/servers/tracing/tempo/default.nix:5]:
```suggestion
  pname = "tempo";
  version = "1.4.1";
```

**Comment by K900** [comment]
Applied all the suggestions.

**Pull request closed without merging**